CenterForOpenScience/osf.io
# Feature/education employment endpoints

## Purpose


Verify and update UserEducation and UserEmployment endpoint functionality, request validation, and testing.

## Changes


* Fix undefined import
* Clean date validation
* Separate `profile_type` and `object_type` in endpoint testing and update test functions to fix broken tests (tests were calling invalid functions/attributes)
* Require `ongoing` parameter to be defined if `start_date` is provided (via jsonschema validation) and add tests accordingly

## QA Notes


The UserEducation and UserEmployment API endpoints should be tested/verified by QA. This includes checking the CRUD (create, read, update, and delete) functionality of the API endpoints, verifying that authorization for the API endpoints is in line with expectations, ensuring that parameter validation is working as expected or that helpful errors are provided.


## Documentation


No, documentation for these endpoints does not currently exist.

## Side Effects


N/A

## Ticket


[Jira Ticket](https://openscience.atlassian.net/browse/ENG-965)
